torch/csrc/autograd/utils/grad_layout_contract.h

```diff
@@ -25,7 +25,7 @@ inline at::Tensor clone_obey_contract(const at::Tensor& new_grad, const at::Tens
     // (1)
     // Does this dicey-looking sequence attach the result to new_grad's
     // history if GradMode::is_enabled()?  Yes, and @alband says it should.
-    return std::move(at::empty_strided(variable.sizes(), variable.strides(),
+    return std::move(new_grad.new_empty_strided(variable.sizes(), variable.strides(),
```

Comment:
is it because `empty_strided` cannot create BatchedTensor?

Reply:
Yes, empty_strided doesn't know how to create a BatchedTensor in this situation because it doesn't accept a tensor as an argument. To make up for it, our solution is to use `new_empty_strided`: new_empty_strided can accept BatchedTensors and if it is passed one, it returns a BatchedTensor.